<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>894</width>
    <height>529</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>894</width>
    <height>529</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>894</width>
    <height>529</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Preferences</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_6">
   <item row="0" column="0">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="minimumSize">
      <size>
       <width>431</width>
       <height>261</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>431</width>
       <height>9999</height>
      </size>
     </property>
     <property name="title">
      <string>Pre-Calibration of Flat Files</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <property name="verticalSpacing">
       <number>8</number>
      </property>
      <item row="9" column="0" colspan="2">
       <widget class="QLabel" name="precalibrationPathDisplay">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;File name is echoed here&lt;br/&gt;second line&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="3">
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>31</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="8" column="0" colspan="2">
       <widget class="QRadioButton" name="FixedPreCalFileRB">
        <property name="toolTip">
         <string>Use fixed bias file specified here for precalibration</string>
        </property>
        <property name="text">
         <string>Fixed Bias or Flat File:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="3">
       <widget class="QRadioButton" name="fixedPedestalRB">
        <property name="toolTip">
         <string>Precalibrate by subtracting given constant pedestal value from every pixel</string>
        </property>
        <property name="text">
         <string>Subtract fixed pedestal amount:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QLineEdit" name="fixedPedestalAmount">
        <property name="toolTip">
         <string>Pedestal value to subtract</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <spacer name="horizontalSpacer_6">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>48</width>
          <height>19</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="2" column="0" colspan="3">
       <widget class="QRadioButton" name="autoPreCalibrationRB">
        <property name="toolTip">
         <string>Automatically choose a matching Bias or Flat file from a directory of Bias files</string>
        </property>
        <property name="text">
         <string>Best Bias or Flat from Directory</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <spacer name="horizontalSpacer_5">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>48</width>
          <height>19</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="4" column="2">
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>45</width>
          <height>19</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="8" column="4">
       <widget class="QPushButton" name="selectPreCalFile">
        <property name="text">
         <string>Set File</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="QRadioButton" name="noPreClalibrationRB">
        <property name="toolTip">
         <string>No precalibration</string>
        </property>
        <property name="text">
         <string>No pre-calibration</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QCheckBox" name="autoBiasOnly">
        <property name="text">
         <string>Dark &amp;&amp; Bias files only</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0" colspan="2">
       <widget class="QLabel" name="autoDirectoryName">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;Directory name echoed here&lt;br/&gt;second line&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="4">
       <widget class="QPushButton" name="setAutoDirectory">
        <property name="toolTip">
         <string>Set the directory containing the calibration Bias files</string>
        </property>
        <property name="text">
         <string>Set Directory</string>
        </property>
        <property name="autoDefault">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="1" colspan="2">
       <widget class="QCheckBox" name="autoRecursive">
        <property name="text">
         <string>Include sub-directories</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <spacer name="horizontalSpacer_7">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>48</width>
          <height>19</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="6" column="1">
       <widget class="QCheckBox" name="displayAutoResultsCB">
        <property name="toolTip">
         <string>Display, in the console, the exposure and temperature match of the selected best calibration file.</string>
        </property>
        <property name="text">
         <string>Display specs of selected file</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QGroupBox" name="groupBox">
     <property name="minimumSize">
      <size>
       <width>431</width>
       <height>281</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>431</width>
       <height>281</height>
      </size>
     </property>
     <property name="title">
      <string>Process Multiple Files at Once, Grouped by Attribute</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="2" column="1">
       <widget class="QLineEdit" name="temperatureGroupBandwidth">
        <property name="maximumSize">
         <size>
          <width>41</width>
          <height>21</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Make groups where temperatures are within this much on either side of the group centre.</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="groupByTemperatureCB">
        <property name="toolTip">
         <string>Process sets of files having different CCD temperatures</string>
        </property>
        <property name="text">
         <string>Group by Temperature, width:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="minimumGroupSize">
        <property name="maximumSize">
         <size>
          <width>41</width>
          <height>21</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Minimum number of files to be processed in a group.</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="QCheckBox" name="groupBySizeCB">
        <property name="toolTip">
         <string>Process sets of files having different dimensions or binnings</string>
        </property>
        <property name="text">
         <string>Group by Size (dimensions and binning)</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QCheckBox" name="ignoreSmallGroupsCB">
        <property name="toolTip">
         <string>Don't process groups that have fewer than the given number of members.</string>
        </property>
        <property name="text">
         <string>Ignore groups with fewer than:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>files</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>51</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="groupByFilterCB">
        <property name="toolTip">
         <string>Process all files with the same filter name in a group</string>
        </property>
        <property name="text">
         <string>Group by Filter</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="CombineMethod">
     <property name="minimumSize">
      <size>
       <width>428</width>
       <height>150</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>428</width>
       <height>150</height>
      </size>
     </property>
     <property name="title">
      <string>Default Combination Algorithm</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QRadioButton" name="combineMeanRB">
        <property name="toolTip">
         <string>Simple mean. Best signal-to-noise, but outliers (satellites etc) will show up.</string>
        </property>
        <property name="text">
         <string>Mean</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">combineMethodGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QLineEdit" name="minMaxNumDropped">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;How many min and max values are dropped?  (1 = drop all instances of the max and all instances of the min; 2 = all instances of max and next-to-max and all instances of 2 min values, etc.)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QLineEdit" name="sigmaThreshold">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Sigma ratio (how many standard deviations the value is from the mean) beyond which data are dropped.  Try 2.0 then experiment.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Sigma rejection threshold</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QRadioButton" name="combineMedianRB">
        <property name="toolTip">
         <string>Simple median. Rejects outliers, but signal-to-noise not as good as with Mean.</string>
        </property>
        <property name="text">
         <string>Median</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">combineMethodGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string># Dropped from each end</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QRadioButton" name="combineMinMaxRB">
        <property name="toolTip">
         <string>Drop min and max values, then Mean. Works well when outliers are hot or black pixels.</string>
        </property>
        <property name="text">
         <string>Min/Max Clip</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">combineMethodGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QRadioButton" name="combineSigmaRB">
        <property name="toolTip">
         <string>Drop pixels more than a given sigma ratio from the mean, then mean the rest. Good rejection of anomalous data.</string>
        </property>
        <property name="text">
         <string>Sigma Clip</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">combineMethodGroup</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="minimumSize">
      <size>
       <width>431</width>
       <height>95</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>431</width>
       <height>151</height>
      </size>
     </property>
     <property name="title">
      <string>Disposition of Input Files after Successful Combine</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="1" column="0">
       <widget class="QRadioButton" name="dispositionSubFolderRB">
        <property name="toolTip">
         <string>Move the input files to a sub-folder (in the same directory as the input files) with the given name</string>
        </property>
        <property name="text">
         <string>Move to Sub-Folder Named:</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">dispositionGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="2" column="0">
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>42</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="subFolderName">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Name for the folder to move the processed files. Optionally this name can contain %d, %t, or %f, which will be replaced by the date, time, and filter name, respectively.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QRadioButton" name="dispositionNothingRB">
        <property name="toolTip">
         <string>Leave the input files alone.</string>
        </property>
        <property name="text">
         <string>Do Nothing</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">dispositionGroup</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QWidget" name="widget" native="true">
     <layout class="QGridLayout" name="gridLayout_5">
      <item row="0" column="0">
       <widget class="QPushButton" name="closeButton">
        <property name="text">
         <string>Close</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>335</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QCheckBox" name="displayAverageADUs">
     <property name="toolTip">
      <string>Display the ADU level for each flat frame.</string>
     </property>
     <property name="text">
      <string>Display average ADUs value of files (slows down opening)</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="dispositionGroup"/>
  <buttongroup name="combineMethodGroup"/>
 </buttongroups>
</ui>
